packages/components/checkbox/src/Checkbox.tsx

```diff
@@ -66,12 +65,45 @@ export const Checkbox = forwardRef<HTMLButtonElement, CheckboxProps>(
           aria-describedby={description}
           aria-invalid={isInvalid}
           onCheckedChange={handleCheckedChange}
+          /**
+           * If the checkbox doesn't have any direct label (children) then we should try to
+           * get an eventual alternative label from FormField.
+           * On last resort, we shouldn't forget to define an aria-label attribute.
+           */
+          aria-labelledby={children ? innerLabelId : field.labelId}
           {...others}
         />
-        {children}
-      </CheckboxLabel>
+
+        {children && (
+          <CheckboxLabel disabled={disabled} htmlFor={id || innerId} id={innerLabelId}>
+            {children}
+          </CheckboxLabel>
+        )}
+      </div>
     )
   }
 )
 
+const getCheckboxAttributes = ({
```

Comment:
I'm not sure about doing this. What's the reasoning of creating functions outside components when they are no being reused somewhere else ?

Reply:
To provide more context, the code had been refactored due to an ESLint warning. The function's complexity exceeded the threshold of the ESLint rule.

And I was the one suggesting @soykje to define the function outside of the component (for the rationale behind this suggestion, see my comment just above)

Reply:
Yes, your fault! :stuck_out_tongue_closed_eyes: Seriously, I'm not a big fan of this syntax too, but as Alex said, this is mainly ESLint faul... we have to respect those rules, or discuss on how we make them evolve :slightly_smiling_face:

Reply:
As discussed, we'll move this function back within `Checkbox`. We'll discuss later about how we live with ESLint :sunglasses:

Reply:
🥲